storage/google_drive/drive_client_wrapper.py

```diff
@@ -0,0 +1,135 @@
+
+import mimetypes
+import os
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = 'https://www.googleapis.com/auth/drive'
+
+drive_service = None
+drive_folder_root = None
+
+def init_client(auth_credentials_path):
+    global drive_service
+    global drive_folder_root
+    store = file.Storage('token.json')
+    creds = store.get()
+    if not creds or creds.invalid:
+        flow = client.flow_from_clientsecrets('credentials.json', SCOPES)
+        creds = tools.run_flow(flow, store)
+    drive_service = build('drive', 'v3', http=creds.authorize(Http()))
+    drive_folder_root = drive_service.files()
+
+def get_root_id():
+    return drive_folder_root.get(fileId='root').execute().get('id')
+
+def get_children(folder_id):
+    """Returns a list with map elements with the following structure:
+    {
+        'name': ''
+        'id': '',
+        'mimeType': '',
+    }
+    """
+    children = []
+    page_token = None
+    while True:
+        response = drive_folder_root.list(
+            q="'{}' in parents".format(folder_id),
+            spaces='drive',
+            fields='nextPageToken, files(id, name, mimeType)',
+            pageToken=page_token).execute()
+        for file in response.get('files', []):
+            children.append(file)
+        page_token = response.get('nextPageToken', None)
+        if page_token is None:
+            break
+    return children
+
+def get_folder_id(name, parent_id):
+    response = drive_folder_root.list(
+        q="name='{}' and '{}' in parents and mimeType='application/vnd.google-apps.folder'".format(name, parent_id),
+        spaces='drive',
+        fields='files(id)').execute()
+    files = response.get('files', [])
+    if (len(files) == 0):
+        raise LookupError('Folder "{}" not found under parent with id {}.'.format(name, parent_id))
+    if (len(files) > 1):
+        raise LookupError('Multiple folders with name "{}" found under parent with id {}.'.format(name, parent_id))
+    return files[0].get('id')
+
+def get_path_id(path):
+    folders = _split_path(path)
+    folder_id = get_root_id()
+    for folder in folders:
+        folder_id = get_folder_id(folder, folder_id)
+    return folder_id
+
+def _split_path(path):
+    folders = []
+    while path != "":
+        path, folder = os.path.split(path)
+        if folder != "":
+            folders.append(folder)
+    folders.reverse()
+    return folders
+
+def update_file(source_file_path, file_id, mimetype):
+    media = MediaFileUpload(source_file_path,
+                            mimetype=mimetype,
+                            resumable=True)
+
+    file = drive_folder_root.update(fileId=file_id,
+                                    media_body=media,
+                                    fields='name').execute()
+    
+    print('File "{}" with ID: {} updated'.format(file.get('name'), file_id))
+
+def create_file(source_file_path, mimetype, parent_folder_id, uploaded_file_name=None):
```

Comment:
'uploaded_file_name' has a different tense - is it the name that will be created in Drive? consider 'target_file_name'

Reply:
parent_folder_id is a drive-api-ism, could we go for 'target_folder'?